src/mscorlib/src/System/Threading/ThreadPool.cs

```diff
@@ -491,6 +491,9 @@ public IThreadPoolWorkItem Dequeue(ThreadPoolWorkQueueThreadLocals tl, ref bool
 
         internal static bool Dispatch()
         {
+            // Put threadpool thread on default context
+            ExecutionContextSwitcher.SetDefaults();
```

Comment:
Threadpool threads don't start on Default context as the VM creates them rather than the managed code